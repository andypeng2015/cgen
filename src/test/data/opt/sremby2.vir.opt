module foo

export function w $foo(w %x) {
@2:
  %4 = lsr.w %x, 0x1f_w ; @8
  %5 = add.w %x, %4 ; @9
  %6 = and.w %5, 0x1_w ; @10
  %7 = sub.w %6, %4 ; @11
  ret %7
}
