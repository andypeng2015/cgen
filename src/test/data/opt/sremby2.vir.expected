module foo

export function w $foo(w %x) {
@0:
  %0 = lsr.w %x, 0x1f_w ; @2
  %1 = add.w %x, %0 ; @3
  %2 = and.w %1, 0x1_w ; @4
  %3 = sub.w %2, %0 ; @5
  ret %3
}
